src/sched/lthread.c

```diff
@@ -733,8 +745,8 @@ int lthread_create(
     lt->attr.stack_size = stack_size;
 
     /* mmap tls image */
-    lt->itlssz = libc.tls_size;
-    if (libc.tls_size)
+    lt->itlssz = (sizeof(lthread_tcb_base) + TLS_ALIGN - 1) & -TLS_ALIGN;
+    if (lt->itlssz)
     {
         if ((lt->itls = (uint8_t*)enclave_mmap(
```

Comment:
If we do the adjustment once here (which is a very cold code path), we can avoid the conditional in the `set_tls_tp` function (which is a hot path, hit on every context switch).